fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1721,35 +1722,48 @@ def _post_backward_hook(self, param: Parameter, *unused: Any) -> None:
         # reductions in post_backward stream.
         self._streams["post_backward"].wait_stream(torch.cuda.current_stream())
         with torch.cuda.stream(self._streams["post_backward"]):
-            orig_grad_data = param.grad.data
+            if param.main_grad is not None and not param.main_grad.eq(0).all():
```

Comment:
Is there another way I can check if main_grad is non zero without doing a CPU sync?

Reply:
We are checking if this is all zeros to skip modules that didn't use main_grad?

Reply:
yes .. because all parameters have `.main_grad`, so not sure how to make sure we are not using the ones that do not have the grads stored in `.main_grad`